crates/extensions/tedge_mqtt_bridge/src/lib.rs

```diff
@@ -83,55 +88,53 @@ impl MqttBridgeActorBuilder {
         local_config.set_manual_acks(true);
         local_config.set_last_will(LastWill::new(
             &health_topic,
-            BridgeHealth::DOWN_PAYLOAD,
+            Status::Down.json(),
             QoS::AtLeastOnce,
             true,
         ));
-        let mut cloud_config = MqttOptions::new(
-            tedge_config.device.id.try_read(tedge_config).unwrap(),
-            tedge_config
-                .c8y
-                .mqtt
-                .or_config_not_set()
-                .unwrap()
-                .host()
-                .to_string(),
-            8883,
-        );
+        local_config.set_clean_session(false);
+
         cloud_config.set_manual_acks(true);
 
-        cloud_config.set_transport(Transport::tls_with_config(tls_config.into()));
-        let topic_prefix = format!("{prefix}/");
         let (local_client, local_event_loop) = AsyncClient::new(local_config, 10);
         let (cloud_client, cloud_event_loop) = AsyncClient::new(cloud_config, 10);
```

Comment:
> `sudo tedge connect c8y --test` returned an unexpected exit code stdout: Sending packets to check connection.

I observe on my laptop the same behavior as on the failing system tests: no JWT token is received when requested on `c8y/s/uat`:

- The message published on `c8y/s/uat` is not even processed by the c8y built-in bridge.
- The built-in bridge is even stuck: processing no more messages after a first batch of 11 messages.
- Changing the `cap` parameter from 10 to some  crazy number 10000 unblocks the built-in bridge
  (this change has to be for the `local_config`, the `cloud_config` and the `bidirectional_channel`).

Reply:
I've now understood and fixed the issue here. The issue was the change to the subscriptions. Previously we were subscribing before connecting, but with this PR, we subscribe upon receiving a `ConnAck`, which means we resubscribe after a dropped connection.

The issue with this is the cloud connection takes a reasonable period of time to come up after starting the mapper, and if during that time 10 messages are sent, this fills up the channel for the cloud event loop. When the cloud then successfully connects, it attempts to subscribe with `recv_client.subscribe_many(&topics).await`. This attempts to send a message to the event loop to process the subscription request, but the event loop is already full, so this results in a deadlock.